actions/rules/action_state_to_metastate.go

```diff
@@ -0,0 +1,472 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/fabric8-services/fabric8-wit/actions/change"
+
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+// ActionStateToMetaState implements the bidirectional mapping between state and column.
+type ActionStateToMetaState struct {
+	Db     application.DB
+	Ctx    context.Context
+	UserID *uuid.UUID
+}
+
+// make sure the rule is implementing the interface.
+var _ Action = ActionStateToMetaState{}
+
+func (act ActionStateToMetaState) containsUUID(s []uuid.UUID, e uuid.UUID) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (act ActionStateToMetaState) contains(s []interface{}, e interface{}) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (act ActionStateToMetaState) removeElement(s []interface{}, e interface{}) []interface{} {
+	for idx, a := range s {
+		if a == e {
+			s = append(s[:idx], s[idx+1:]...)
+			// we don't return here as there may be multiple copies of e in s.
+		}
+	}
+	return s
+}
+
+// difference returns the differences between two slices. It returns two slices containing
+// the added and removed elements compared from old and new arguments.
+func (act ActionStateToMetaState) difference(old []interface{}, new []interface{}) ([]interface{}, []interface{}) {
```

Comment:
Please write a test for this function. I cannot find any.

Reply:
Fixed. d867145d20d15824a896d20858306042e2865136